pkg/apis/kubermatic/v1/resource_quota.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:generate=true
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name="Age",type="date"
+
+// ResourceQuota specifies the amount of cluster resources a project can use.
+type ResourceQuota struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ResourceQuotaSpec   `json:"spec,omitempty"`
+	Status ResourceQuotaStatus `json:"status,omitempty"`
+}
+
+// ResourceQuotaSpec describes the desired state of a resource quota.
+type ResourceQuotaSpec struct {
+	// QuotaSubject specifies to which entity the quota applies to.
+	QuotaSubject QuotaSubject `json:"quotaSubject"`
```

Comment:
Wouldn't `subject` be enough?` "ResourceQuota.quotaSubject" sounds doublespeaky.